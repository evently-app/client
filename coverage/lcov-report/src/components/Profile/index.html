<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/Profile</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> src/components/Profile
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">40.35% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>23/57</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">20.83% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>5/24</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">29.41% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>5/17</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">40.74% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>22/54</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="Dial.js"><a href="Dial.js.html">Dial.js</a></td>
	<td data-value="29.55" class="pic low"><div class="chart"><div class="cover-fill" style="width: 29%;"></div><div class="cover-empty" style="width:71%;"></div></div></td>
	<td data-value="29.55" class="pct low">29.55%</td>
	<td data-value="44" class="abs low">13/44</td>
	<td data-value="11.11" class="pct low">11.11%</td>
	<td data-value="18" class="abs low">2/18</td>
	<td data-value="30.77" class="pct low">30.77%</td>
	<td data-value="13" class="abs low">4/13</td>
	<td data-value="30.95" class="pct low">30.95%</td>
	<td data-value="42" class="abs low">13/42</td>
	</tr>

<tr>
	<td class="file medium" data-value="index.js"><a href="index.js.html">index.js</a></td>
	<td data-value="76.92" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 76%;"></div><div class="cover-empty" style="width:24%;"></div></div></td>
	<td data-value="76.92" class="pct medium">76.92%</td>
	<td data-value="13" class="abs medium">10/13</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="6" class="abs medium">3/6</td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="4" class="abs low">1/4</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="12" class="abs medium">9/12</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Apr 14 2019 20:53:00 GMT-0400 (Eastern Daylight Time)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
<script src="../../../block-navigation.js"></script>
</body>
</html>
