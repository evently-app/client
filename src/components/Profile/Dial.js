import React, { Component } from "react";
import { View, PanResponder, Animated, StyleSheet } from "react-native";
import { AnimatedCircularProgress } from "react-native-circular-progress";
import Haptics from "react-native-haptic-feedback";

// determines how often the Dial ticks for haptics
const genTick = fill => {
	return Math.round(fill * 7);
};

const PADDING_HORIZONTAL = 20;
const PADDING_VERTICAL = 20;

const SIZE = 116;

class Dial extends Component {
	constructor(props) {
		super(props);
		this.state = {
			fill: props.fill
		};
		this.activated = new Animated.Value(1);
		this.tick = genTick(this.state.fill);
		this.handleLayout = this.handleLayout.bind(this);
	}

	shouldComponentUpdate(nextProps, nextState) {
		if (nextState.fill != this.state.fill) {
			return true;
		} else if (nextState.activated != this.state.activated) {
			return true;
		} else {
			return false;
		}
	}

	refView = ref => (this.view = ref);

	componentWillMount() {
		this._panResponder = PanResponder.create({
			onMoveShouldSetResponderCapture: () => true,
			onMoveShouldSetPanResponderCapture: () => true,

			onPanResponderGrant: (e, gestureState) => {},

			onPanResponderMove: (e, { x0, y0, dx, dy }) => {
				// animate dial fill

				const panPoint = { x: x0 + dx, y: y0 + dy };

				// pattern 1

				const onRightSide = panPoint.x > this.center.x;

				// get lengths of implicit lines for calculation
				const centerTop = this.center.y - this.centerTop.y;
				const centerOut = Math.sqrt(
					Math.pow(this.center.y - panPoint.y, 2) +
						Math.pow(this.center.x - panPoint.x, 2)
				);
				const topOut = Math.sqrt(
					Math.pow(this.centerTop.y - panPoint.y, 2) +
						Math.pow(this.centerTop.x - panPoint.x, 2)
				);

				// use line lengths and law of cosine to calculate implied angle from drag
				const angle = Math.acos(
					(Math.pow(centerTop, 2) +
						Math.pow(centerOut, 2) -
						Math.pow(topOut, 2)) /
						(2 * centerOut * centerTop)
				);

				// interpret angle as fill amount
				let fill = onRightSide
					? (angle / Math.PI) * 0.7 - 0.15
					: 0.5 + (1 - angle / Math.PI) * 0.6;
				if (fill < 0) fill = 0;
				if (fill > 1) fill = 1;
				this.setState({ fill });

				// pattern 2

				// measure distance from center
				// const distance = Math.sqrt(
				// 	Math.pow(this.center.y - panPoint.y, 2) +
				// 		Math.pow(this.center.x - panPoint.x, 2)
				// );
				// let fill = (distance - 40) / 300;
				// if (fill < 0) fill = 0;
				// if (fill > 1) fill = 1;
				// this.setState({ fill });

				// haptic if past next tick
				if (this.tick != genTick(fill)) {
					Haptics.trigger("impactLight");
					this.tick = genTick(fill);
				}
			},

			onPanResponderRelease: (e, { vx, vy }) => {
				// activation animation
				if (this.state.fill > 0) {
					Animated.timing(this.activated, {
						toValue: 1,
						duration: 100
					}).start();
				} else {
					Animated.timing(this.activated, {
						toValue: 0,
						duration: 100
					}).start();
				}

				// publish change
				if (this.props.onChange) {
					this.props.onChange(this.state.fill);
				}
			}
		});
	}

	handleLayout() {
		this.view.measure((fx, fy, width, height, px, py) => {
			this.center = {
				x: fx + PADDING_HORIZONTAL + SIZE / 2,
				y: py + PADDING_VERTICAL + SIZE / 2
			};
			this.centerTop = {
				x: fx + PADDING_HORIZONTAL + SIZE / 2,
				y: py + PADDING_VERTICAL
			};
		});
	}

	render() {
		return (
			<View
				ref={this.refView}
				style={styles.wrapper}
				onLayout={this.handleLayout}
				{...this._panResponder.panHandlers}
			>
				<AnimatedCircularProgress
					size={SIZE}
					width={10}
					rotation={0}
					fill={this.state.fill * 100}
					tintColor="rgba(110,10,234,0.95)"
					backgroundColor="rgba(110,10,234,0.30)"
				/>
				<View style={styles.innerCircleWrapper}>
					<Animated.View
						style={[
							styles.innerCircle,
							{
								opacity: this.activated.interpolate({
									inputRange: [0, 1],
									outputRange: [0.2, 1]
								})
							}
						]}
					/>
				</View>
				{!!this.props.title && (
					<Animated.Text
						style={[
							styles.title,
							{
								opacity: this.activated.interpolate({
									inputRange: [0, 1],
									outputRange: [0.6, 1]
								})
							}
						]}
					>
						{this.props.title}
					</Animated.Text>
				)}
			</View>
		);
	}
}

const styles = StyleSheet.create({
	wrapper: {
		flexDirection: "column",
		padding: 20
	},
	title: {
		fontSize: 16,
		fontWeight: "700",
		color: "white",
		fontFamily: "Avenir Next",
		paddingTop: 14,
		textAlign: "center"
	},
	innerCircle: {
		width: 80,
		height: 80,
		borderRadius: 40,
		borderWidth: 15,
		borderColor: "white"
	},
	innerCircleWrapper: {
		position: "absolute",
		top: 0,
		left: 0,
		right: 0,
		bottom: 35,
		justifyContent: "center",
		alignItems: "center"
	}
});

export default Dial;
