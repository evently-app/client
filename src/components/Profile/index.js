import React, { Component } from "react";
import { View, Text } from "react-native";

import { SCREEN_WIDTH, SCREEN_HEIGHT } from "../../lib/constants";

class Profile extends Component {
	render() {
		return (
			<View
				style={{
					flex: 1,
					width: SCREEN_WIDTH,
					height: SCREEN_HEIGHT,
					backgroundColor: "rgba(0, 255, 0, 0.2)",
					justifyContent: "center",
					alignItems: "center"
				}}
			>
				<Text>Profile</Text>
			</View>
		);
	}
}

export default Profile;
