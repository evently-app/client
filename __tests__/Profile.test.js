import "react-native";
import React from "react";
import Profile from "../src/components/Profile";
import renderer from "react-test-renderer";

test("Profile renders correctly", () => {
	const tree = renderer.create(<Profile />).toJSON();
	expect(tree).toMatchSnapshot();
});
