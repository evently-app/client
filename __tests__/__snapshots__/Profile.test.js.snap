// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dial renders correctly 1`] = `
<View
  onLayout={[Function]}
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  style={
    Object {
      "flexDirection": "column",
      "padding": 20,
    }
  }
>
  <AnimatedCircularProgress
    backgroundColor="rgba(110,10,234,0.30)"
    duration={500}
    easing={[Function]}
    fill={NaN}
    prefill={0}
    rotation={0}
    size={116}
    tintColor="rgba(110,10,234,0.95)"
    width={10}
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "bottom": 35,
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <AnimatedComponent
      style={
        Array [
          Object {
            "borderColor": "white",
            "borderRadius": 40,
            "borderWidth": 15,
            "height": 80,
            "width": 80,
          },
          Object {
            "opacity": 1,
          },
        ]
      }
    />
  </View>
  <AnimatedComponent
    style={
      Array [
        Object {
          "color": "white",
          "fontFamily": "Avenir Next",
          "fontSize": 16,
          "fontWeight": "700",
          "paddingTop": 14,
          "textAlign": "center",
        },
        Object {
          "opacity": 1,
        },
      ]
    }
  >
    test dial
  </AnimatedComponent>
</View>
`;

exports[`Profile renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-start",
      "flex": 1,
      "flexDirection": "column",
      "height": 1334,
      "justifyContent": "flex-start",
      "paddingBottom": 20,
      "paddingTop": 40,
      "width": 750,
    }
  }
>
  <Header
    style={
      Object {
        "marginHorizontal": 20,
      }
    }
  >
    Preferences
  </Header>
  <SubHeader
    style={
      Object {
        "marginHorizontal": 20,
      }
    }
  >
    I like events that are...
  </SubHeader>
  <View
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "space-around",
        "paddingTop": 20,
      }
    }
  >
    <Dial
      fill={0}
      key="dial-lit"
      onChange={[Function]}
      title="Lit"
    />
    <Dial
      fill={0}
      key="dial-active"
      onChange={[Function]}
      title="Active"
    />
    <Dial
      fill={0}
      key="dial-relaxing"
      onChange={[Function]}
      title="Relaxing"
    />
    <Dial
      fill={0}
      key="dial-cultural"
      onChange={[Function]}
      title="Cultural"
    />
  </View>
</View>
`;
